.github/dependabot.yaml

```diff
@@ -18,3 +19,40 @@ updates:
     - "release-note-none"
     - "ok-to-test"
   open-pull-requests-limit: 10
+
+# The list below needs to be maintained manually with every branch we support.
```

Comment:
nit: group package-ecosystem changes in one block i.e. before the github-actions block or moving the first one here
nit: squash commits

cc @dzjiang91, if the PR gets created for a release branch I think that we could do the diff with the one created for master to see if there are changes i.e. to see if cherrypicking the change on master to the release branches is the same as what the bot creates.